### DISCUSSION TITLE
Commands not found while executing makefile

### DISCUSSION THREAD

**Discussion by silviocross** [adafruit/Adafruit_nRF52_Bootloader#278] [Q&A]
Hi everybody,
I am trying to build the bootloader from source since I need to enable ANT using the S340 SoftDevice.
Additionally to the official documentation I am currently following this guide: https://blogarak.wordpress.com/2020/03/15/s340-softdevice-adafruit-nrf52840-express-feather/ 

Unfortunately, as soon as I run the makefile many errors appear as "command not found"

`C:\Adafruit_nRF52_Bootloader>make BOARD=mdk_nrf52840_dongle all combinehex`
`/usr/bin/sh: cut: command not found`
`/usr/bin/sh: paste: command not found`
`/usr/bin/sh: mkdir: command not found`
`make: *** [Makefile:385: _build/build-mdk_nrf52840_dongle] Error 127`

I'm running Windows 10, GNU ARM Toolchain and nRF5x Command Line Tools are already installed.

Can anybody guide me to a possible solution?

Thank you very much in advance
Silvio

**Comment by hathach** [comment]
it is built-in command of linux shell, maybe trying to install cygwin, msys would help.

**Comment by amowry** [comment]
I can confirm that it is possible to build on Windows using cygwin.

**Reply by MIKHANYA** [reply]
Can you please suggest or direct me what needs to be done? Or what am I doing wrong?
- Used this article as a step-by-step guide.
 https://blogarak.wordpress.com/2020/03/15/s340-softdevice-adafruit-nrf52840-express-feather/
- installed python,
- nRF5x Command Line Tools (for Windows and Linux on cygwin)
- GCC tool chain https://developer.arm.com/downloads/-/gnu-rm (for Windows and Linux on cygwin)

When I'm trying to create bootloader file `make BOARD=feather_nrf52840_sense all`
have compiler error `-bash: make: command not found`
Please help, just want to generate bootloader file with some edits


![image](https://github.com/user-attachments/assets/6e288e2e-75fd-4f3c-a0d8-361ceff43397)

**Comment by JzPyro** [comment]
Go into your system setings after adding gcc and make sure you add the bin folder of the extracted folder to the system path environment variables. Here is the instructions for Win 11:
1. In the File Explorer Right Click on "This PC" and select "Properties"
2. In the settings page that pops up select "Advanced System Settings" under device specifications
3. At the bottom of the popup click "Environment variables" 
4. In the bottom section scroll down until you see the "Path" Environment Variable, Double Click it or Click it to highlight it and click edit.
5. Finally Click "New" and type in the Path to the bin folder for the installed tool. For me this was in "C:\MinGW\bin" but i installed MinGW instead of the traditional GCC.
